bfvmm/include/memory_manager/page_table_entry_x64.h

```diff
@@ -59,155 +76,257 @@ class page_table_entry_x64
 
     /// Present
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @return true if this entry is present, false otherwise
     ///
     bool present() const noexcept;
 
     /// Set Present
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @param enabled true if the entry is present, false otherwise
     ///
     void set_present(bool enabled) noexcept;
 
     /// Read / Write
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @return true if this entry is read/write, false otherwise
     ///
     bool rw() const noexcept;
 
     /// Set Read / Write
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @param enabled true if the entry is read / write, false otherwise
     ///
     void set_rw(bool enabled) noexcept;
 
     /// User / Supervisor
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @return true if this entry is visible to userspace, false otherwise
     ///
     bool us() const noexcept;
 
     /// Set User / Supervisor
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @param enabled true if the entry is visible to userspace, false
     ///     otherwise
     ///
     void set_us(bool enabled) noexcept;
 
     /// Write-Through
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @return true if this entry is write-through, false otherwise
     ///
     bool pwt() const noexcept;
 
     /// Set Write-Through
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @param enabled true if the entry is write-through, false otherwise
     ///
     void set_pwt(bool enabled) noexcept;
 
     /// Cache Disable
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @return true if this entry's cache is disabled, false otherwise
     ///
     bool pcd() const noexcept;
 
     /// Set Cache Disable
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @param enabled true if the entry's cache is disabled, false
     ///     otherwise
     ///
     void set_pcd(bool enabled) noexcept;
 
     /// Accessed
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @return true if this entry has been accessed, false otherwise
     ///
     bool accessed() const noexcept;
 
     /// Set Accessed
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @param enabled true if the entry has been accessed, false
     ///     otherwise
     ///
     void set_accessed(bool enabled) noexcept;
 
     /// Dirty
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @return true if this entry is dirty, false otherwise
     ///
     bool dirty() const noexcept;
 
     /// Set Dirty
     ///
+    /// @expects none
+    /// @ensures none
+    ///
     /// @param enabled true if the entry is dirty, false otherwise
     ///
     void set_dirty(bool enabled) noexcept;
 
+    /// Page Size
```

Comment:
Yeah, that's out of the Intel manual. PS == Page Size, and it basically means, does this represent a page (i.e. this page size == #bits) or does this represent another set of page tables (i.e. page size - bits). I hate it too